samples/src/App.js

```diff
@@ -32,41 +32,142 @@ function App() {
   const [localMicrophoneTrack, setLocalMicrophoneTrack] = useState();
   const [localDisplayTracks, setLocalDisplayTracks] = useState();
 
+  const [bnrEffect, setBnrEffect] = useState();
+  const [blurEffect, setBlurEffect] = useState();
+  const [vbgEffect, setVirtualBackgroundEffect] = useState();
+  const [videoEffect, setVideoEffect] = useState();
+
+  const [trackEvents, setTrackEvents] = useState([]);
+
+  const updateEvents = (event) => {
+    setTrackEvents(trackEvents => [...trackEvents,{message: event}]);
+  }
+
+  const enableBnr = (enabled) =>{
+    const options = {
+      audioContext,
+      processorUrl,
+      legacyProcessorUrl,
+      mode,
+    };
+    effect = new NoiseReductionEffect(options);
+
+  }
+
+  const enableVirtualBackground = (enabled) => {
+
+    if(!videoEffect) {
+      effect = new VirtualBackgroundEffect({
+        mode: "IMAGE",
+        bgVideoUrl: VIDEO_URL,
+      });
+
+      localCameraTrack.addEffect("virtualBackground", effect)
+    }
+
+  }
+
+  const enableVideoBackground = (enabled) => {
+    effect = new VirtualBackgroundEffect({
+      mode: "VIDEO",
+      bgVideoUrl: VIDEO_URL,
+    });
+  }
+
+  const enableBlurBackground = async (enabled) => {
+    let effect = null;
+    if(!blurEffect) {
+      effect = new VirtualBackgroundEffect({
+        mode: `BLUR`,
+        blurStrength: `STRONG`,
+        quality: `HIGH`,
+      });
+      await localCameraTrack.addEffect("blur", effect);
+      setBlurEffect(effect)
```

Comment:
does it make sense to create this even if enabled is false?

Reply:
if so, maybe it should just be created by default instead of in the first call to `enableBlurBackground`?

Reply:
its up to the developer does not make a difference , creating it ahead or later

Reply:
Right, but the point of the sample is to show the idiomatic way of doing it, so I think we should make sure the example is clear.

Reply:
with react it gets confusing , thought it would be better to keep the effect creation and management in a single place 

we are not going to use this demo for developers , There will be a separate doc on the SDK side for the usage 

i can move the VBG creation on the useState when the app loads if you want